Mathlib/SetTheory/Ordinal/Principal.lean

```diff
@@ -99,9 +99,11 @@ theorem principal_nfp_blsub₂ (op : Ordinal → Ordinal → Ordinal) (o : Ordin
     ((fun o' => blsub₂.{u, u, u} o' o' (@fun a _ b _ => op a b))^[n] o) with h h
   · use n + 1
     rw [Function.iterate_succ']
+    dsimp
```

Comment:
Why do we need `dsimp`s now?

Reply:
@digama0, could you comment here?

(This PR was me trying to get Mathlib green against this PR, I admit I've forgotten, if I ever understood at all, the effect of the underlying Lean 4 change.)

Reply:
The effect of the lean 4 change is that when faced with problems of the form `foo x =?= bar y`, it will now sometimes unfold `foo` where it used to unfold `bar` or vice versa. This may cause timing differences or possible timeouts, although I wouldn't expect any particular bias to whether things get better or worse, maybe a little bit better since the heuristic is supposed to have better luck than picking randomly... I don't know off hand why this `dsimp` would be needed.